samples/net/sockets/echo_async/src/socket_echo.c

```diff
@@ -70,36 +76,72 @@ void pollfds_del(int fd)
 
 int main(void)
 {
-	int serv;
-	struct sockaddr_in bind_addr;
+	int res;
+	int serv4, serv6;
+	struct sockaddr_in bind_addr4 = {
+		.sin_family = AF_INET,
+		.sin_port = htons(4242),
+		.sin_addr = {
+			.s_addr = htonl(INADDR_ANY),
+		},
+	};
+	struct sockaddr_in6 bind_addr6 = {
+		.sin6_family = AF_INET6,
+		.sin6_port = htons(4242),
+		.sin6_addr = IN6ADDR_ANY_INIT,
+	};
+
+	serv4 = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+	res = bind(serv4, (struct sockaddr *)&bind_addr4, sizeof(bind_addr4));
+	if (res == -1) {
+		printf("Cannot bind IPv4, errno: %d\n", errno);
+	}
 
-	serv = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+	serv6 = socket(AF_INET6, SOCK_STREAM, IPPROTO_TCP);
+	#ifdef IPV6_V6ONLY
+	/* For Linux, we need to make socket IPv6-only to bind it to the
+	 * same port as IPv4 socket above.
+	 */
+	int TRUE = 1;
+	setsockopt(serv6, IPPROTO_IPV6, IPV6_V6ONLY, &TRUE, sizeof(TRUE));
+	#endif
+	res = bind(serv6, (struct sockaddr *)&bind_addr6, sizeof(bind_addr6));
+	if (res == -1) {
+		printf("Cannot bind IPv6, errno: %d\n", errno);
+	}
 
-	bind_addr.sin_family = AF_INET;
-	bind_addr.sin_addr.s_addr = htonl(INADDR_ANY);
-	bind_addr.sin_port = htons(4242);
-	bind(serv, (struct sockaddr *)&bind_addr, sizeof(bind_addr));
+	nonblock(serv4);
+	nonblock(serv6);
+	listen(serv4, 5);
+	listen(serv6, 5);
 
-	nonblock(serv);
-	pollfds_add(serv);
+	pollfds_add(serv4);
+	pollfds_add(serv6);
 
-	listen(serv, 5);
+	printf("Listening on port 4242...\n");
 
 	while (1) {
-		struct sockaddr_in client_addr;
+		struct sockaddr_storage client_addr;
 		socklen_t client_addr_len = sizeof(client_addr);
 		char addr_str[32];
 
-		poll(pollfds, pollnum, -1);
+		res = poll(pollfds, pollnum, -1);
+		if (res == -1) {
+			printf("poll error: %d\n", errno);
+			continue;
+		}
+
 		for (int i = 0; i < pollnum; i++) {
 			if (!(pollfds[i].revents & POLLIN)) {
 				continue;
 			}
 			int fd = pollfds[i].fd;
-			if (fd == serv) {
-				int client = accept(serv, (struct sockaddr *)&client_addr,
+			if (i < 2) {
+				/* If server socket */
+				int client = accept(fd, (struct sockaddr *)&client_addr,
 						    &client_addr_len);
-				inet_ntop(client_addr.sin_family, &client_addr.sin_addr,
+				void *addr = &((struct sockaddr_in *)&client_addr)->sin_addr;
```

Comment:
Well, it is - declared at the start of block together with "int client" before it, and both are initialized at the same time. Adding blank line.